ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt

```diff
@@ -143,37 +144,49 @@ class SubmissionRepository @Inject constructor(
             }
     }
 
-    fun deleteSubmission(submission: Submission): @Cold Completable =
-        applyAndEnqueue(
-            builder()
-                .setJob(submission.job)
-                .setSubmissionId(submission.id)
-                .setResponseDeltas(ImmutableList.of())
-                .setType(Mutation.Type.DELETE)
-                .setSyncStatus(SyncStatus.PENDING)
-                .setSurveyId(submission.survey.id)
-                .setLocationOfInterestId(submission.locationOfInterest.id)
-                .setClientTimestamp(Date())
-                .setUserId(authManager.currentUser.id)
-                .build()
-        )
+    fun deleteSubmission(submission: Submission): @Cold Completable {
+        return if (submission.locationOfInterest.id == null) {
+            Completable.error(DataStoreException("Missing LOI id."))
+        } else {
+            applyAndEnqueue(
+                builder()
+                    .setJob(submission.job)
+                    .setSubmissionId(submission.id)
+                    .setResponseDeltas(ImmutableList.of())
+                    .setType(Mutation.Type.DELETE)
+                    .setSyncStatus(SyncStatus.PENDING)
+                    .setSurveyId(submission.survey.id)
+                    .setLocationOfInterestId(
+                        submission.locationOfInterest.id!!
+                    )
+                    .setClientTimestamp(Date())
+                    .setUserId(authManager.currentUser.id)
+                    .build()
+            )
+        }
+    }
 
     fun createOrUpdateSubmission(
         submission: Submission, responseDeltas: ImmutableList<ResponseDelta>, isNew: Boolean
-    ): @Cold Completable =
-        applyAndEnqueue(
-            builder()
-                .setJob(submission.job)
-                .setSubmissionId(submission.id)
-                .setResponseDeltas(responseDeltas)
-                .setType(if (isNew) Mutation.Type.CREATE else Mutation.Type.UPDATE)
-                .setSyncStatus(SyncStatus.PENDING)
-                .setSurveyId(submission.survey.id)
-                .setLocationOfInterestId(submission.locationOfInterest.id)
-                .setClientTimestamp(Date())
-                .setUserId(authManager.currentUser.id)
-                .build()
-        )
+    ): @Cold Completable {
+        return if (submission.locationOfInterest.id == null) {
+            Completable.error(DataStoreException("Missing LOI id."))
+        } else {
+            applyAndEnqueue(
+                builder()
+                    .setJob(submission.job)
+                    .setSubmissionId(submission.id)
+                    .setResponseDeltas(responseDeltas)
+                    .setType(if (isNew) Mutation.Type.CREATE else Mutation.Type.UPDATE)
+                    .setSyncStatus(SyncStatus.PENDING)
+                    .setSurveyId(submission.survey.id)
+                    .setLocationOfInterestId(submission.locationOfInterest.id!!)
```

Comment:
Same as above.

Reply:
Thanks! Will address in a follow up.

Reply:
oh- scratch that, this field is already non-null in this PR, these checks are redundant. Removed.